### DISCUSSION TITLE
French plurals

### DISCUSSION THREAD

**Discussion by nijel** [WeblateOrg/weblate#13279] [Q&A]
Hi, we're a little confused, specifically about the French plurals.

We were using Weblate 4.14 with i18next JSON v4, and had singular/plural suffixes `_one` and `_many`. Now, we upgraded to the latest Weblate 5.8.4, and I thought that [this would introduce a new plural rule for French](https://github.com/WeblateOrg/weblate/issues/7888), such that there are `_one`, `_many`, and `_other`. However, I still see the rule `n > 1` for French, and I'm not sure if that refers to `_many` or `_other` now.

 @nijel could you be so kind and explain how (French) plurals are currently handled in Weblate, based on which CLDR version, using integer or decimal rules?

We also have scripts to synchronize and sort our JSON translation files, and we need to know the plural suffices for those scrips, so it would be great if we knew where Weblate is taking them from. For the script, we're extracting the suffixes from the cardinal integer rules from https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/plurals.json, but that doesn't seem to match Weblate, because for French that results in `_one`, `_many`, and `_other`.

_Originally posted by @darioseidl in https://github.com/WeblateOrg/weblate/issues/7520#issuecomment-2541499555_

**Comment by nijel** [comment]
@darioseidl This was really off-topic at the issue, let's follow-up here. 

The plural formula should be different with three plurals, see for example https://hosted.weblate.org/translate/stretchly/stretchly/fr/?checksum=f2e178e87c7c6166.

It will be updated once the files are parsed again with the new version.

**Comment by darioseidl** [comment]
@nijel Thank you for your answer! You're right, it wasn't on topic there, and I only needed to trigger parsing the translation files again by committing a change to them to update to the new plural rules. Now we're seeing the correct plural rules with three cases. 

If anyone's interested, I can share our script that synchronizes the JSON translation files and takes the plural rules into account ...
